File: 1319-unique-number-of-occurrences/1319-unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
       HashMap<Integer, Integer> frequencyMap = new HashMap<>();
        
        for (int num : arr) {
            frequencyMap.put(num, frequencyMap.getOrDefault(num, 0) + 1);
        }
        HashSet<Integer> occurrences = new HashSet<>(frequencyMap.values());
        return frequencyMap.size() == occurrences.size(); 
    }
}

File: 1319-unique-number-of-occurrences/1319-unique-number-of-occurrences.go
func uniqueOccurrences(arr []int) bool {
    frequencyMap := make(map[int]int)
	for _, num := range arr {
		frequencyMap[num]++
	}
	occurrenceSet := make(map[int]bool)

	for _, freq := range frequencyMap {
		if occurrenceSet[freq] {
			return false
		}
		occurrenceSet[freq] = true
	}
	return true
}